ffs/coreipfs/coreipfs.go

```diff
@@ -106,6 +107,12 @@ func (ci *CoreIpfs) Store(ctx context.Context, c cid.Cid) (int, error) {
 	if err != nil {
 		return 0, fmt.Errorf("getting stats of cid %s: %s", c, err)
 	}
+	if err := ci.ensurePinsetCache(ctx); err != nil {
```

Comment:
Could this just be done on startup?

Reply:
Yep, that's an option. But I tried to avoid adding load to bootstrapping and just load it on first use. But this can perfectly be moved to `New`.

Reply:
Cool, makes sense. I guess the only downside of having it here is that there will one (the first after start) lucky user that gets a slow response time.

Reply:
Yeah, that's a good point.
The other option is spinning a go routine on `New`... "nobody" pays a cost. :P

Reply:
Oh yeah, sounds good 💯